Description: never allow launching of untrusted launchers.  Implements the
 Ubuntu "Execute-Permission Bit Required" from
 https://wiki.ubuntu.com/SecurityTeam/Policies#Execute-Permission%20Bit%20Required
Author: Kees Cook <kees@ubuntu.com>

Index: nemo-3.3.92/src/nemo-mime-actions.c
===================================================================
--- nemo-3.3.92.orig/src/nemo-mime-actions.c	2012-03-07 20:15:47.000000000 +0100
+++ nemo-3.3.92/src/nemo-mime-actions.c	2012-03-20 11:19:28.591758144 +0100
@@ -1553,12 +1553,14 @@
 			      "text", primary,
 			      "secondary-text", secondary,
 			      NULL);
+		if (nemo_file_get_ctime(file) < 1256817600L ) {
 		gtk_dialog_add_button (GTK_DIALOG (dialog),
 				       _("_Launch Anyway"), RESPONSE_RUN);
 		if (nemo_file_can_set_permissions (file)) {
 			gtk_dialog_add_button (GTK_DIALOG (dialog),
 					       _("Mark as _Trusted"), RESPONSE_MARK_TRUSTED);
 		}
+		}
 		gtk_dialog_add_button (GTK_DIALOG (dialog),
 				       GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL);
 		gtk_dialog_set_default_response (GTK_DIALOG (dialog), GTK_RESPONSE_CANCEL);
Index: nemo-3.3.92/libnemo-private/nemo-file.c
===================================================================
--- nemo-3.3.92.orig/libnemo-private/nemo-file.c	2012-03-20 11:19:28.519758140 +0100
+++ nemo-3.3.92/libnemo-private/nemo-file.c	2012-03-20 11:19:33.587758385 +0100
@@ -4892,6 +4892,12 @@
 	return file->details->mtime;
 }
 
+time_t
+nemo_file_get_ctime (NemoFile *file)
+{
+	return file->details->ctime;
+}
+
 
 static void
 set_attributes_get_info_callback (GObject *source_object,
Index: nemo-3.3.92/libnemo-private/nemo-file.h
===================================================================
--- nemo-3.3.92.orig/libnemo-private/nemo-file.h	2011-12-12 17:31:15.000000000 +0100
+++ nemo-3.3.92/libnemo-private/nemo-file.h	2012-03-20 11:19:28.603758144 +0100
@@ -168,6 +168,7 @@
 gboolean                nemo_file_can_get_size                      (NemoFile                   *file);
 goffset                 nemo_file_get_size                          (NemoFile                   *file);
 time_t                  nemo_file_get_mtime                         (NemoFile                   *file);
+time_t                  nemo_file_get_ctime                         (NemoFile                   *file);
 GFileType               nemo_file_get_file_type                     (NemoFile                   *file);
 char *                  nemo_file_get_mime_type                     (NemoFile                   *file);
 gboolean                nemo_file_is_mime_type                      (NemoFile                   *file,
